release/mkrelease.js

```diff
@@ -0,0 +1,306 @@
+            process.exit(-1);
+        }
+        var lastReleaseDate = body.published_at;
+        console.log("Fetching PRs merged since " + lastReleaseDate);
+        gitHubRequest("https://api.github.com/search/issues?q=type:pr+is:merged+repo:microsoft/azure-pipelines-agent+merged:>=" + lastReleaseDate + "&sort=closed_at&order=asc", { json: true }, function (err, resp, body) {
+            editReleaseNotesFile(body, callback);
+        });
+    });
+}
+
+function editReleaseNotesFile(body, callback)
+{
+    var releaseNotesFile = __dirname + '/../releaseNote.md';
+    var existingReleaseNotes = fs.readFileSync(releaseNotesFile);
+    var newPRs = [];
+    body.items.forEach(function (item) {
+        newPRs.push(' - ' + item.title + ' (#' + item.number + ')');
+    });
+    var newReleaseNotes = newPRs.join("\n") + "\n\n" + existingReleaseNotes;
+    var editorCmd = process.env.EDITOR + ' ' + releaseNotesFile;
+    console.log(editorCmd);
+    if (opt.options.dryrun)
+    {
+        console.log("Found the following PRs = %o", newPRs);
+    }
+    else
+    {
+        fs.writeFileSync(releaseNotesFile, newReleaseNotes);
+        try
+        {
+            cp.execSync(process.env.EDITOR + ' ' + releaseNotesFile, {
+                stdio: [process.stdin, process.stdout, process.stderr]
+            });
+        }
+        catch (err)
+        {
+            console.log(err.message);
+            process.exit(-1);
+        }
+    }
+    callback();
+}
+
+function versionifySync(template, destination, version)
+{
+    try
+    {
+        var data = fs.readFileSync(template, 'utf8');
+        data = data.replace(/<AGENT_VERSION>/g, version);
+        console.log("Generating " + destination);
+        fs.writeFileSync(destination, data);
+    }
+    catch(e)
+    {
+        console.log('Error:', e.stack);
+    }
+}
+
+function createIntegrationFiles(newRelease, callback)
+{
+    mkpath(integrationDir, function (err) {
+        if (err) throw err;
+        cp.execSync("rm  -rf " + integrationDir + "/PublishVSTSAgent-*");
+        versionifySync(__dirname + "/../src/Misc/InstallAgentPackage.template.xml",
+            integrationDir + "/InstallAgentPackage.xml",
+            newRelease
+        );
+        var agentVersionPath=newRelease.replace('.', '-');
+        var publishDir = integrationDir + "/PublishVSTSAgent-" + agentVersionPath
+            mkpath(publishDir, function (err) {
+                if (err) throw err;
+                versionifySync(__dirname + "/../src/Misc/PublishVSTSAgent.template.ps1",
+                publishDir + "/PublishVSTSAgent-" + agentVersionPath + ".ps1",
+                newRelease
+            );
+            versionifySync(__dirname + "/../src/Misc/UnpublishVSTSAgent.template.ps1",
+                publishDir + "/UnpublishVSTSAgent-" + agentVersionPath + ".ps1",
+                newRelease
+            );
+            callback();
+        });
+    });
+}
+
+function execInForground(command, directory)
```

Comment:
Spelling nit: `execInForeground`

Reply:
Any reason we need to specify that its happening in the foreground? Really this is just `execSync` with a little directory/opt support.

Reply:
agreed, this api needs some doc.

Reply:
my guess is that it has a "dryrun" option

Reply:
The issue is that execSync by default does not connect stdin, stdout, stderr to the same as the running node process. For some of the git commands, I wanted the output to go to the terminal. For the spawning of the editor to work in the terminal, I needed stdin and stdout hooked up. I also wanted to have the dry run option in one place. I'll add some comments :)